pbsmetrics/config/metrics.go

```diff
@@ -33,14 +35,84 @@ func NewMetricsEngine(cfg *config.Configuration, adapterList []openrtb_ext.Bidde
 		)
 		// Influx is not added to the engine list as goMetrics takes care of it already.
 	}
+	if cfg.Metrics.Prometheus.Port != 0 {
+		// Set up the Prometheus metrics.
+		proMetrics := prometheusmetrics.NewMetrics(cfg.Metrics.Prometheus)
+		engineList = append(engineList, proMetrics)
+	}
 
 	// Now return the proper metrics engine
 	if len(engineList) > 1 {
-		return &engineList
+		returnEngine.metricsEngine = &engineList
 	} else if len(engineList) == 1 {
-		return engineList[0]
+		returnEngine.metricsEngine = engineList[0]
+	} else {
+		returnEngine.metricsEngine = &DummyMetricsEngine{}
 	}
-	return &DummyMetricsEngine{}
+
+	return &returnEngine
+}
+
+// DetailedMetricsEngine is a MultiMetricsEngine that preserves links to unerlying metrics engines.
+type DetailedMetricsEngine struct {
+	metricsEngine     pbsmetrics.MetricsEngine
+	GoMetrics         *pbsmetrics.Metrics
+	PrometheusMetrics *prometheusmetrics.Metrics
+}
+
+// RecordRequest passthrough
```

Comment:
The tricky thing here is that pbsmetrics.MetricsEngine is an interface, not a struct. It can hold 4 different concrete types. So I don't see how this construction could work, and if it did, how would we fill the interface with the actual type?

Reply:
```
		return &DetailedMetricsEngine{
			MetricsEngine:     &engineList,
			PrometheusMetrics: proMetrics,
			GoMetrics:         goMetrics,
		}
```

Reply:
related: I found a last-minute bug in this code... you're never actually setting `GoMetrics` and `PrometheusMetrics` in the `DetailedMetricsEngine`. You should fix that regardless.

Reply:
Good catch, though I fixed it differently. :)